internal/util/k8s.go

```diff
@@ -104,3 +105,26 @@ func GetPodDetails(ctx context.Context, kubeClient clientset.Interface) (*PodInf
 		Labels:    pod.GetLabels(),
 	}, nil
 }
+
+// map of all the supported Group/Kinds for the backend. At the moment, only
+// core services are supported, but to provide support to other kinds, it is
+// enough to add entries to this map.
+var backendRefSupportedGroupKinds = map[string]interface{}{
```

Comment:
If this is meant to be used as "set" then the value type of this map can be `struct{}`

Reply:
Yes, the value may be `struct{}` or `interface{}`, I chose `interface{}` to stick with the current approach that we currently have in many other places around the code. I wonder if there is any difference in terms of the performance of space consumption.

Reply:
The only solution is then to measure!

```go
package main

import "testing"

func BenchmarkStruct(b *testing.B) {
	m := make(map[int]struct{})
	for i := 0; i < b.N; i++ {
		m[i] = struct{}{}
	}
}

func BenchmarkInterface(b *testing.B) {
	m := make(map[int]interface{})
	for i := 0; i < b.N; i++ {
		m[i] = nil
	}
}
```

Drumroll....... 🥁 

```
go test -count 5 -v -benchmem -bench .
goos: darwin
goarch: arm64
pkg: mapset
BenchmarkStruct
BenchmarkStruct-10              17631705               124.2 ns/op            45 B/op          0 allocs/op
BenchmarkStruct-10              17228527               120.7 ns/op            46 B/op          0 allocs/op
BenchmarkStruct-10              18327208               118.0 ns/op            44 B/op          0 allocs/op
BenchmarkStruct-10              18146235               114.6 ns/op            44 B/op          0 allocs/op
BenchmarkStruct-10              16389720               121.6 ns/op            49 B/op          0 allocs/op
BenchmarkInterface
BenchmarkInterface-10           11918082               125.3 ns/op            85 B/op          0 allocs/op
BenchmarkInterface-10           11904559               125.7 ns/op            85 B/op          0 allocs/op
BenchmarkInterface-10           11752041               125.6 ns/op            86 B/op          0 allocs/op
BenchmarkInterface-10           11338368               127.3 ns/op            88 B/op          0 allocs/op
BenchmarkInterface-10           11628338               126.7 ns/op            86 B/op          0 allocs/op
PASS
ok      mapset  18.791s
```

Reply:
Well, looks like `struct{}` won the race :) I'll change it